test/sequential/test-child-process-emfile.js

```diff
@@ -9,15 +9,25 @@ if (common.isWindows) {
   return;
 }
 
+var open_fds = [];
+
 for (;;) {
   try {
-    fs.openSync(__filename, 'r');
+    open_fds.push(fs.openSync(__filename, 'r'));
   } catch (err) {
     assert(err.code === 'EMFILE' || err.code === 'ENFILE');
     break;
   }
 }
 
+if (process.env.LSAN_OPTIONS) {
+  // Close two fds because LSan needs one but before the testrunner grabs one
+  if (open_fds.length >= 2) {
+    fs.closeSync(open_fds.pop());
+    fs.closeSync(open_fds.pop());
+  }
+}
```

Comment:
Can't you do this unconditionally at the end of the test?  There's less failure mode if the LSAN_OPTIONS sniffing can be removed.